src/Datadog.Trace/Agent/NamedPipes/TraceRequest.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using Datadog.Trace.Vendors.MessagePack;
+
+namespace Datadog.Trace.Agent.NamedPipes
+{
+    internal class TraceRequest
+    {
+        public Dictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();
```

Comment:
I suggest using `List<KeyValuePair<string, string>>` (initialized to the expected capacity, if possible) instead of `Dictionary` since we don't need the constant-time lookups (we only add headers and then iterate the entire collection once) or unique keys (http headers can appear multiple times in a request).